### PR TITLE
feat: support set operations in flink

### PR DESCRIPTION
**Pull request by poulinjulien**
This PR adds support for [set operations](https://nightlies.apache.org/flink/flink-docs-release-1.16/docs/dev/table/sql/queries/set-ops/) in Flink.

Please review carefully, as this "language" is very new to me :-)

### CODE REVIEW

File: pegjs/flinksql.pegjs [hunk 1]
```diff
@@ -257,8 +259,13 @@ multiple_stmt
...
     }
 
+set_op
+  = u:(KW_UNION / KW_INTERSECT / KW_EXCEPT) __ s:KW_ALL? {
+    return s ? `${u.toLowerCase()} ${s.toLowerCase()}` : `${u.toLowerCase()}`
+  }
+
 union_stmt
-  = head:select_stmt tail:(__ KW_UNION __ KW_ALL? __ select_stmt)* __ ob: order_by_clause? __ l:limit_clause? {
+  = head:select_stmt tail:(__ set_op? __ select_stmt)* __ ob: order_by_clause? __ l:limit_clause? {
```
Comment (poulinjulien) [inline pegjs/flinksql.pegjs:268]: I'm not sure if that `?` is necessary. Tests pass without it.

```suggestion
  = head:select_stmt tail:(__ set_op __ select_stmt)* __ ob: order_by_clause? __ l:limit_clause? {
```

**Comment by taozhi8833998** [comment]
@poulinjulien hi, do you have any future updates on this pr? I can wait until you think this pr is ready for review.

**Comment by poulinjulien** [comment]
@taozhi8833998 hi, the pr is ready, I just rebased it and removed the `?`, as I suspect it was not correct.

**Comment by taozhi8833998** [comment]
> @taozhi8833998 hi, the pr is ready, I just rebased it and removed the `?`, as I suspect it was not correct.

ok, thanks, I will review it now.

**Comment by taozhi8833998** [comment]
@poulinjulien merged it.

**Comment by poulinjulien** [comment]
@taozhi8833998 thank you! I have a few more lined up, should I submit all of them at once?

**Comment by taozhi8833998** [comment]
> @taozhi8833998 thank you! I have a few more lined up, should I submit all of them at once?

submit it by feature, so code review would be more efficient.

**Pull request merged**